### ISSUE TITLE
Feature request: hide sensitive data

### ISSUE THREAD

**Issue by linuxmail** [Icinga/icingaweb2-module-director#1651]
Hi,
it would be nice to hide all sensitive fields. It doesn't feel good to see the password/snmp values in cleartext for variables etc.

**Comment by dbu1986** [comment]
Have you considered using the protected_customvars option in the monitoring module configuration?

https://icinga.com/docs/icingaweb2/latest/modules/monitoring/doc/03-Configuration/

**Comment by linuxmail** [comment]
hi Thomas,

that would be awesome.

@dbu1986 no, that I have configured and they are working. It is realy the config preview. In many applications I can click a show/hide button, to make them unreadable.

cu denny

**Comment by Thomas-Gelf** [comment]
@linuxmail: I was referring just to the form field, not to the configuration preview. Preview could also be addressed, but that's more complicated. You can implement config diff by rendering some hash value, but I cannot use a Field implementation at that point in time. Also, multiple differing Fields might have been defined for `community` on different templates, one uses it for SNMP communities and the other template for Social Media nonsense.

I'd suggest to revoke the `director/showconfig` permission from those who should not see sensitive data. Still, I like the show/hide proposal. This requires for a global `sensitive vars` setting like in the monitoring module, and it could default to what has been defined over there.

**Comment by linuxmail** [comment]
hi,

@Thomas-Gelf its not who has permissions, but hide it from eyes behind you :-), so for me its O.K to just hide the sensitives behind ********* ... and make it visible, if I want.

**Comment by Thomas-Gelf** [comment]
Sure, I got it. Problem is that I still need to be able to somehow show THAT the sensitive string has been changed when showing a configuration diff. That's tricky, because you know what is sensitive information before rendering. But you have to keep the original value in the rendered config, otherwise the diff would fail. Replacing everything with *** means no diff. So information needs to be hidden AFTER being rendered, but then you have a text blob with no meaning.

Now, there is already some regex-Magic going on to place links into a rendered configuration, and matching `vars.<special word>` is easy. But what if `vars.credential` looks like this:

```
    vars.credential = {
        username = "monitoring"
        password = "icinga"
        ...
    }
```

Good luck with regular expressions.

**Comment by meyerf99** [comment]
Are there any news if a Data type "Password" can be added to the Icinga2 Director Data Fields (variables)?
I would appreciate it very much because currently as already described by @linuxmail , passwords are plain text visible on Web UI.

**Comment by Thomas-Gelf** [comment]
This should have been addressed with #1872 and been released with v1.7.0. It doesn't touch the config preview, so either do not grant that permission or do not click "Preview" when someone stands behind you ;)
